src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1401,12 +1428,13 @@ public Conversion ClassifyImplicitExtensionMethodThisArgConversion(BoundExpressi
 
             if ((object)sourceType != null)
             {
+                Debug.Assert(!IncludeNullability); // PROTOTYPE(NullableReferenceTypes): Add test.
```

Comment:
Please confirm the `Debug.Assert(!IncludeNullability)` assertions you added in a couple of places. 
Are those "not yet reachable" or will they remain unreachable? 
If the latter, we should probably throw `UnreachableException` and consider leaving a comment to explain why it is unreachable.

Reply:
These asserts should fail although the code paths are not yet reachable. I've changed the comments to be more specific.

---
In reply to: [164223440](https://github.com/dotnet/roslyn/pull/23981#discussion_r164223440) [](ancestors = 164223440)